<a href="https://colab.research.google.com/github/batuhanyndny/notebooks/blob/master/Scikit_MLPClassifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from sklearn.neural_network import MLPClassifier
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split, KFold
from sklearn.metrics import confusion_matrix

import warnings
warnings.filterwarnings('ignore')

In [ ]:
digits = load_digits()
X = digits.data
y = digits.target

kf = KFold(n_splits=5)

best_clf = None
best_score = 0
best_indices = None
for train_index, test_index in kf.split(X):
  X_train, X_test = X[train_index], X[test_index]
  y_train, y_test = y[train_index], y[test_index]
  clf = MLPClassifier(random_state=42, max_iter=10, activation='logistic', solver='adam').fit(X_train, y_train)
  score = clf.score(X_test, y_test)
  if (score > best_score):
    best_clf = clf
    best_score = score
    best_indices = {"train": train_index, "test": test_index}


y_pred = best_clf.predict(X_test)
conf_matrix = confusion_matrix(y_test, y_pred)
print(f'Best classifier score: {best_score}\nConfusion Matrix: {conf_matrix}')

Best classifier score: 0.9415041782729805
Confusion Matrix: [[34  0  0  0  1  0  0  0  0  0]
 [ 0 27  0  0  1  0  0  0  0  8]
 [ 0  0 33  1  0  0  0  0  0  0]
 [ 0  1  0 30  0  1  1  3  1  0]
 [ 0  0  0  0 34  0  0  0  3  0]
 [ 0  1  0  0  0 36  0  0  0  0]
 [ 0  2  0  0  0  0 35  0  0  0]
 [ 0  0  0  0  0  0  0 35  1  0]
 [ 0  2  0  1  1  2  0  2 21  4]
 [ 0  0  0  2  0  3  0  1  1 30]]


# LOGS
`solver: sgd` , `activation: relu` : 0.9192200557103064 

`solver: sgd` , `activation: tanh` : 0.7855153203342619

`solver: sgd` , `activation: logistic` : 0.7103064066852368

`solver: adam` , `activation: relu` : 0.9275766016713092

`solver: adam` , `activation: tanh` : 0.935933147632312

`solver: adam` , `activation: logistic` : 0.9415041782729805
